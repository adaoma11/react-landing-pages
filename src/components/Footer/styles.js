import styled, { css } from 'styled-components';

export const Footer = styled.footer`
  ${({ theme, $hasBg }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    height: 150px;
    background-color: ${$hasBg ? theme.colors.primary : theme.colors.white};
  `}
`;
