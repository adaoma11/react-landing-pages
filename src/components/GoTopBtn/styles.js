import styled, { css } from 'styled-components';

export const Link = styled.a`
  ${({ theme }) => css`
    display: flex;
    width: 40px;
    height: 40px;
    position: fixed;
    bottom: 2rem;
    right: 2rem;
    color: ${theme.colors.white};
    background-color: ${theme.colors.primary};
    border-radius: 5px;
    opacity: 0.8;
  `}
`;
