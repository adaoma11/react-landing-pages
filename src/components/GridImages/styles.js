import styled from 'styled-components';

export const GridContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  width: 100%;
  gap: 20px;
  margin-top: 50px;
`;

export const GridItem = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 0.5s ease-in-out;

  &:hover {
    transform: scale(1.2) rotate(5deg);
  }
`;
