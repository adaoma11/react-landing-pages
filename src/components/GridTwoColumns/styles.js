import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 1fr;
    align-items: center;
    gap: 30px;

    @media ${theme.medias.medium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const TextContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: 30px;
`;

export const Image = styled.div`
  ${({ $imgSrc }) => css`
    display: block;
    min-height: 500px;
    background-image: url(${$imgSrc});
    background-repeat: no-repeat;
    background-position: center;
    background-size: 60%;
  `}
`;
