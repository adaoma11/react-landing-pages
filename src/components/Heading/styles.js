import styled, { css } from 'styled-components';

export const Title = styled.h1`
  ${({ theme, $dark, size, $upperCase }) => css`
    color: ${$dark ? theme.colors.primary : theme.colors.white};

    font-size: ${theme.fonts.sizes[size]};

    ${size == 'xlarge' &&
    css`
      @media ${theme.medias.small} {
        font-size: ${theme.fonts.sizes.large};
      }
    `}

    text-transform: ${$upperCase ? 'uppercase' : 'none'}
  `}
`;
