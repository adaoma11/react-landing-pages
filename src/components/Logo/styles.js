import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.heading};
    font-size: ${theme.fonts.sizes.large};
    font-weight: bold;
    color: ${theme.colors.primary};
    text-decoration: none;

    > img {
      min-height: 40px;
      max-height: 50px;
      max-width: 150px;
      min-width: 40px;
    }
  `}
`;
