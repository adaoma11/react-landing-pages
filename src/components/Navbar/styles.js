import styled, { css } from 'styled-components';
import {
  Menu as OpenSvg,
  Close as CloseSvg,
} from '@styled-icons/material-outlined';

export const Nav = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-flow: row;
    justify-content: space-between;
    align-items: center;
    position: sticky;
    top: 0;
    width: 100%;
    max-width: 1200px;
    margin: 0 auto;
    padding: 10px 0 10px 0;
    background-color: ${theme.colors.white};

    @media (max-width: 1230px) {
      padding: 0px 20px;
    }

    @media ${theme.medias.small} {
      align-items: start;
      justify-content: space-around;

      > button {
        margin: 20px 0px 20px 0px;
      }

      > a > img {
        margin: 15px 0px 15px 0px;
      }
    }
  `}
`;

export const Menu = styled.menu`
  ${({ theme, $visible }) => css`
    display: flex;
    flex-flow: row wrap;
    list-style: none;
    transition: ${$visible
      ? 'height 0.8s linear, opacity 1s linear 0.4s'
      : 'height 0.5s linear'};

    @media ${theme.medias.small} {
      visibility: ${$visible ? 'visible' : 'hidden'};
      opacity: ${$visible ? '1' : '0'};
      height: ${$visible ? '100vh' : '0'};
      flex-flow: column nowrap;
      justify-content: center;
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    display: none;
    margin: 10px 10px 0 0;
    border: none;
    background: none;
    cursor: pointer;

    @media ${theme.medias.small} {
      display: block;
    }
  `}
`;

export const openBtn = styled(OpenSvg)`
  ${({ theme }) => css`
    width: 35px;
    height: 35px;
    color: ${theme.colors.primary};
  `};
`;

export const closeBtn = styled(CloseSvg)`
  ${({ theme }) => css`
    width: 35px;
    height: 35px;
    color: ${theme.colors.primary};
  `};
`;
