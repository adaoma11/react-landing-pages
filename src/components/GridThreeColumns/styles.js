import styled from 'styled-components';
import { Title as GridItemTitle } from '../Heading/styles';

export const GridContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  counter-reset: grid-counter;
  width: 100%;
  gap: 50px;
`;

export const GridItem = styled.div`
  margin-top: 80px;

  ${GridItemTitle} {
    position: relative;
    left: 4rem;
    margin-bottom: 30px;
  }

  ${GridItemTitle}::before {
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 5rem;
    top: -2rem;
    left: -4rem;
    transform: rotate(7deg);
  }
`;
