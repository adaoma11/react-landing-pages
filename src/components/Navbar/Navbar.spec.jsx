import { renderTheme } from '../../styles/renderTheme';
import { fireEvent, screen } from '@testing-library/react';
import { Navbar } from '.';

const mock = {
  logoData: {
    text: 'Teste',
    imgSrc: 'vite.svg',
    href: '#',
  },
  links: [
    {
      children: 'Link 1',
      href: '#',
      newTab: false,
    },
    {
      children: 'Link 2',
      href: '#',
      newTab: false,
    },
    {
      children: 'Link 3',
      href: '#',
      newTab: false,
    },
    {
      children: 'Link 4',
      href: '#',
      newTab: false,
    },
    {
      children: 'Link 5',
      href: '#',
      newTab: false,
    },
  ],
};

describe('<Navbar />', () => {
  it('should render a navbar with all its children', () => {
    renderTheme(<Navbar {...mock} />);
    expect(screen.getByRole('img')).toBeInTheDocument();
    expect(screen.getAllByRole('link')).toHaveLength(6);
  });

  it('should shrink on small screens', () => {
    renderTheme(<Navbar {...mock} />);
    expect(screen.getByLabelText('main menu')).toHaveStyleRule(
      'flex-flow',
      'column nowrap',
      {
        media: '(max-width:  768px)',
      },
    );
  });

  it('should open and close menu on small screens', () => {
    renderTheme(<Navbar {...mock} />);

    const openBtn = screen.getByLabelText('open menu');
    const menu = screen.getByLabelText('main menu');

    expect(menu).toHaveStyleRule('opacity', '0', {
      media: '(max-width:  768px)',
    });

    fireEvent.click(openBtn);
    expect(menu).toHaveStyleRule('opacity', '1', {
      media: '(max-width:  768px)',
    });
    expect(menu).toHaveStyleRule('height', '100vh', {
      media: '(max-width:  768px)',
    });

    const closeBtn = screen.getByLabelText('close menu');

    fireEvent.click(closeBtn);
    expect(menu).toHaveStyleRule('opacity', '0', {
      media: '(max-width:  768px)',
    });
    expect(menu).toHaveStyleRule('height', '0', {
      media: '(max-width:  768px)',
    });
  });

  it('should close the menu when clicking on one of its links on small screens', () => {
    renderTheme(<Navbar {...mock} />);

    const openBtn = screen.getByLabelText('open menu');
    const menu = screen.getByLabelText('main menu');

    fireEvent.click(openBtn);
    expect(menu).toHaveStyleRule('opacity', '1', {
      media: '(max-width:  768px)',
    });

    fireEvent.click(menu.firstChild);
    expect(menu).toHaveStyleRule('opacity', '0', {
      media: '(max-width:  768px)',
    });
  });

  it('should match snapshot', () => {
    renderTheme(<Navbar {...mock} />);
    expect(screen.getByRole('navigation')).toMatchSnapshot();
  });
});
