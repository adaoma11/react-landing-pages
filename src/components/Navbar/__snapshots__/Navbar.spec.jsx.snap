// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should match snapshot 1`] = `
.c5 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  display: flex;
  flex-flow: row;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  max-width: 1200px;
  margin: 0 auto;
  padding: 10px 0;
}

.c2 {
  display: flex;
  flex-flow: row wrap;
  list-style: none;
  transition: height 0.5s linear;
}

.c4 {
  display: none;
  margin: 10px 10px 0 0;
  border: none;
  background: none;
  cursor: pointer;
}

.c6 {
  width: 35px;
  height: 35px;
  color: #0A1128;
}

.c3 {
  display: block;
  padding: 25px;
  font-family: 'Poppins',sans-serif;
  font-size: 1rem;
  color: #0A1128;
  text-decoration: none;
  position: relative;
}

.c3::after {
  content: '';
  position: absolute;
  background: #DC143C;
  left: 25%;
  bottom: 0.75rem;
  width: 50%;
  height: 3px;
  transform: scale(0);
  transition: all 0.4s ease-in-out;
}

.c3:hover::after {
  transform: scale(1);
}

.c1 {
  font-family: 'Montserrat',sans-serif;
  font-size: 1.5rem;
  font-weight: bold;
  color: #0A1128;
  text-decoration: none;
}

.c1 >img {
  min-height: 40px;
  max-height: 50px;
  max-width: 150px;
  min-width: 40px;
}

@media (max-width: 1230px) {
  .c0 {
    padding: 10px 20px;
  }
}

@media (max-width: 768px) {
  .c0 {
    min-height: 100px;
    align-items: center;
    justify-content: center;
  }

  .c0 >button {
    position: absolute;
    top: 20px;
    right: 30px;
  }

  .c0 >a {
    position: absolute;
    top: 30px;
    left: 30px;
  }
}

@media (max-width: 768px) {
  .c2 {
    visibility: hidden;
    opacity: 0;
    height: 0;
    flex-flow: column nowrap;
    justify-content: center;
    text-align: center;
  }
}

@media (max-width: 768px) {
  .c4 {
    display: block;
  }
}

<nav
  class="c0"
>
  <a
    class="c1"
    href="#"
  >
    <img
      alt="Teste"
      src="vite.svg"
    />
  </a>
  <menu
    aria-label="main menu"
    class="c2"
  >
    <li>
      <a
        class="c3"
        href="#"
        target="_self"
      >
        Link 1
      </a>
    </li>
    <li>
      <a
        class="c3"
        href="#"
        target="_self"
      >
        Link 2
      </a>
    </li>
    <li>
      <a
        class="c3"
        href="#"
        target="_self"
      >
        Link 3
      </a>
    </li>
    <li>
      <a
        class="c3"
        href="#"
        target="_self"
      >
        Link 4
      </a>
    </li>
    <li>
      <a
        class="c3"
        href="#"
        target="_self"
      >
        Link 5
      </a>
    </li>
  </menu>
  <button
    aria-label="open menu"
    class="c4"
  >
    <svg
      aria-hidden="true"
      class="c5 c6"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
</nav>
`;
